### ISSUE TITLE
How do you host multiple functions in the same project, and call them?

### ISSUE THREAD

**Issue by imaji** [corvus-dotnet/Corvus.Testing#239]
I've got an Azure Function project with two functions in. One's an HTTP Trigger, the other a Timer Trigger.

Using the Azure Function specflow bits, I can point at the project, and the functions start up fine. I can navigate to the local address, and get the "You functions are running" page.

However, when I try calling the HTTP Function using it's defined route, I get 404 back. If I then remove the Timer function, make the exact same call to the same URL as before, it now resolves function, and enters the Run method.

So basically, how do I host multiple functions in a single project? Should I be able to just make the call to the expected route using the HttpClient and expect it to work?

Thanks!

**Comment by jongeorge1** [comment]
Hi, yes, I'd expect it to just work.

To test this, I added a timer triggered function to the demo project, set to execute every 5 seconds and include the last execution time in the message returned by the GET endpoint. I also tweaked the specs to take account of, and test for this, and it all looks to be working fine. You can see the code here - I'd be interested to know if the specs all pass for you.

https://github.com/corvus-dotnet/Corvus.Testing/tree/feature/239-timer-trigger-issue

Also worth checking what version of `azure-functions-core-tools` you have installed globally. For info, I'm currently on 3.0.3442.

If the specs pass, but you still see the problem when using your own functions, are you able to share your code (or a stripped down version) so I can see the issue in action?

**Comment by imaji** [comment]
Ok cool, I'll try modding the demo functions and see if I can replicate my problem.

I realised I'd missed off an important detail, I'm actually using the recently released .net 5 Azure Functions.

**Comment by imaji** [comment]
I'm on 3.0.3442 as well :)

**Comment by jongeorge1** [comment]
Ah, yes, that is an important detail! We haven't done any testing against the .NET 5 version of Azure Functions. Please let us know how you get on. Thanks!

**Comment by imaji** [comment]
Ok, this repo (unmodified) doesn't pass running the tests locally :(

The function call is returning "<!DOCTYPE html><html><head><title>Your Azure Function App is up and running.</title><style type="text/css">...", which is the default page when the functions are up and running.

**Comment by imaji** [comment]
Any joy your end?

**Comment by dustintran333** [comment]
> Ok, this repo (unmodified) doesn't pass running the tests locally :(
> 
> The function call is returning "<title>Your Azure Function App is up and running.</title><style type="text/css">...", which is the default page when the functions are up and running.

I got the same issue and the demo tests are all failing.

**Comment by idg10** [comment]
I just ensured my local clone of the repo was up to date, and then from the `Solutions` folder ran:

```
dotnet build -c Release
dotnet test -c Release
```

I got the output shown below. They all say "Passed!" including the two demo projects.

```
Passed!  - Failed:     0, Passed:     6, Skipped:     0, Total:     6, Duration: 252 ms - Corvus.Testing.SpecFlow.Specs.dll (net8.0)

Passed!  - Failed:     0, Passed:    18, Skipped:     0, Total:    18, Duration: 50 s - Corvus.Testing.AzureFunctions.Xunit.Demo.dll (net8.0)

Passed!  - Failed:     0, Passed:    42, Skipped:     0, Total:    42, Duration: 5 m 52 s - Corvus.Testing.AzureFunctions.SpecFlow.Demo.dll (net8.0)
```

So I'll need some more information if I'm going to be able to reproduce the problem.

What OS are you running on?

What version of the Azure Functions emulator do you have? When I run `func --version` it shows:

```
4.0.5455
```

If you have an old version you can update with this command:

```
npm update -g azure-functions-core-tools@4
```

What error messages are you seeing from the failing tests?

Also, is it possible that security software (e.g., Windows Firewall) has prevented the local Azure functions emulator (`func.exe`) from accepting incoming requests?
